servers/src/test/java/tachyon/master/permission/FileSystemPermissionCheckerTest.java

```diff
@@ -15,35 +15,310 @@
+
+    blockMaster.start(true);
+
+    TachyonConf conf = new TachyonConf();
+    conf.set(Constants.SECURITY_AUTHORIZATION_PERMISSION_ENABLED, "true");
+    conf.set(Constants.SECURITY_AUTHORIZATION_PERMISSION_SUPERGROUP, TEST_SUPER_GROUP);
+    MasterContext.reset(conf);
+    sTree.initializeRoot(TEST_PERMISSION_STATUS_SUPER);
+
+    // verify initialization
+    verifyPermissionChecker(true, TEST_PERMISSION_STATUS_SUPER.getUserName(), TEST_SUPER_GROUP);
+
+    // verify initializing root twice
+    Inode root = sTree.getInodeByPath(new TachyonURI("/"));
+    sTree.initializeRoot(TEST_PERMISSION_STATUS_SUPER);
+    verifyPermissionChecker(true, root.getUserName(), TEST_SUPER_GROUP);
+
+    // build file structure
+    createFileAndDirs();
+  }
+
+  private static void createFileAndDirs() throws Exception {
+    sTree.createPath(new TachyonURI(TEST_DIR_FILE_URI), NESTED_FILE_OPTIONS);
+    sTree.createPath(new TachyonURI(TEST_FILE_URI), FILE_OPTIONS);
+    sTree.createPath(new TachyonURI(TEST_WEIRD_FILE_URI), WEIRD_FILE_OPTIONS);
+
+    verifyInodesList(TEST_DIR_FILE_URI.split("/"),
+        sTree.collectInodes(new TachyonURI(TEST_DIR_FILE_URI)));
+    verifyInodesList(TEST_FILE_URI.split("/"),
+        sTree.collectInodes(new TachyonURI(TEST_FILE_URI)));
+    verifyInodesList(TEST_WEIRD_FILE_URI.split("/"),
+        sTree.collectInodes(new TachyonURI(TEST_WEIRD_FILE_URI)));
+    verifyInodesList(new String[]{"", "testDir"},
+        sTree.collectInodes(new TachyonURI(TEST_NOT_EXIST_URI)));
+  }
+
+  private static void verifyInodesList(String[] expectedInodes, List<Inode> inodes) {
+    String[] inodesName = new String[inodes.size()];
+    for (int i = 0; i < inodes.size(); i++) {
+      inodesName[i] = inodes.get(i).getName();
+    }
+
+    Assert.assertArrayEquals(expectedInodes, inodesName);
+  }
+
+  private static void verifyPermissionChecker(boolean enabled, String owner, String group) {
+    Assert.assertEquals(enabled, Whitebox.getInternalState(FileSystemPermissionChecker.class,
         "sPermissionCheckEnabled"));
-    Assert.assertEquals("user1", Whitebox.getInternalState(FileSystemPermissionChecker.class,
+    Assert.assertEquals(owner, Whitebox.getInternalState(FileSystemPermissionChecker.class,
         "sFileSystemOwner"));
-    Assert.assertEquals("group1", Whitebox.getInternalState(FileSystemPermissionChecker.class,
+    Assert.assertEquals(group, Whitebox.getInternalState(FileSystemPermissionChecker.class,
         "sFileSystemSuperGroup"));
   }
 
-  // TODO(dong): add tests for permission check methods
+  @Test
+  public void testFileSystemOwner() throws Exception {
+    checkSelfPermission(TEST_USER_ADMIN, FileSystemAction.ALL, TEST_DIR_FILE_URI);
+  }
+
+  @Test
+  public void testFileSystemSuperGroup() throws Exception {
+    checkSelfPermission(TEST_USER_SUPERGROUP, FileSystemAction.ALL, TEST_DIR_FILE_URI);
+  }
+
+  @Test
+  public void testSelfCheckSuccess() throws Exception {
```

Comment:
could you rename this method to `selfCheckSuccessTest` to match tachyon code style. Also the other test names.